File: CS/DevExtremeMvcApp1/Controllers/OrdersController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;
using DevExtreme.AspNet.Data;
using DevExtreme.AspNet.Mvc;
using System.Net.Http.Formatting;
using DevExtremeMvcApp1.Models;

namespace DevExtremeMvcAppGridTest.Controllers {
    public class OrdersController : ApiController {
        [HttpGet]
        public HttpResponseMessage Get(DataSourceLoadOptions loadOptions) {
            var list = SampleData.Orders;
            return Request.CreateResponse(DataSourceLoader.Load(list, loadOptions));
        }
        [HttpPut]
        public object Put(FormDataCollection form) {
            //add your code to modify database here
            return Request.CreateResponse(HttpStatusCode.OK);
        }
        [HttpPost]
        public object Post(FormDataCollection form) {
            //add your code to modify database here
            return Request.CreateResponse(HttpStatusCode.OK);
        }
        [HttpDelete]
        public object Delete(FormDataCollection form) {
            //add your code to modify database here
            return Request.CreateResponse(HttpStatusCode.OK);
        }
    }
}

File: VB/Controllers/OrdersController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Net
Imports System.Net.Http
Imports System.Web.Http
Imports DevExtremeMvcAppGridTest.Models
Imports DevExtreme.AspNet.Data
Imports DevExtreme.AspNet.Mvc
Imports System.Net.Http.Formatting

Namespace DevExtremeMvcAppGridTest.Controllers
    Public Class OrdersController
        Inherits ApiController
        <HttpGet>
        Public Function [Get](ByVal loadOptions As DataSourceLoadOptions) As HttpResponseMessage
            Dim list = SampleData.Orders
            Return Request.CreateResponse(DataSourceLoader.Load(list, loadOptions))
        End Function
        <HttpPut>
        Public Function [Put](ByVal form As FormDataCollection) As HttpResponseMessage
            'add your code to modify database here
            Return Request.CreateResponse(HttpStatusCode.OK)
        End Function
        <HttpPost>
        Public Function [Post](ByVal form As FormDataCollection) As HttpResponseMessage
            'add your code to modify database here
            Return Request.CreateResponse(HttpStatusCode.OK)
        End Function
        <HttpDelete>
        Public Function [Delete](ByVal form As FormDataCollection) As HttpResponseMessage
            'add your code to modify database here
            Return Request.CreateResponse(HttpStatusCode.OK)
        End Function
    End Class
End Namespace